docs/branching.md

```diff
@@ -29,9 +29,39 @@ We follow [Trunk Based Development](https://trunkbaseddevelopment.com/) as a bra
 
 Charon is set up to create a release with Github Actions triggered by a tag. To create a new release:
 
-1. Ensure the build is green and that all tests are passing
-2. Identify the version of the release, e.g. `v0.1.2` (note the `v`).
-3. Push a PR that bumps charon version global variable, see example https://github.com/ObolNetwork/charon/pull/312
-4. Merge above PR, checkout latest main, and tag it: `git tag <version> && git push --tags`
-5. Double-check the generated release and changelog: `https://github.com/ObolNetwork/charon/releases/tag/<version>`
-6. Maybe generate a changelog manually and/or update the release: `go run testutil/genchangelog/main.go`
+The important aspects of the release process are:
+- Releases are cut from release branches, not the main branch. Release branches are named `main-v0.X`.
+- Release candidates, `v0.X.Y-rc[1-99]`, are created for each patch release from commits in the release branch. They are thoroughly tested both internally and externally before a release is created.
+- Critical patches and fixes to releases are cherry-picked from main to the release branch.
+- The Charon binary version, `charon version`, are inferred from git tags at build time using `ldflags`, not hardcoded app/version versions.
+- Hardcoded Charon app/version is only used to indicate branch type and major version, `v0.X-rc` for release branches or `v0.Y-dev` for main branch.
+
+The process to follow for the next v0.16.0 release is the following:
```

Comment:
can omit using exact numbers like `v0.16.0` and replace with `v0.X.Y` in the following section

Reply:
Not sure about that, I think concrete examples is better

Reply:
then it would have been great to mention it as a like `For example, ...` it might get confusing IMO from reader's perspective